unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -209,6 +242,16 @@ public boolean equals(final Object other) {
 
   @Override
   public int hashCode() {
-    return Arrays.hashCode(bytes);
+    long result = 1;
```

Comment:
Just copied from Arrays.hashCode(), I think it doesn't matter much.

For caching, we didn't really use UTF8String as the key, it's often to use InternalRow as key, so it's better to cache the hashCode inside InternalRow.